pkg/config/secret/agent_test.go

```diff
@@ -142,22 +152,22 @@ func testAddWithParser(t *testing.T) {
 
 	checkValueAndErr := func(expected int, e error) {
 		t.Helper()
-		select {
-		case v := <-vals:
-			if v != expected {
-				t.Errorf("expected value to get updated to %d but got updated to %d", expected, v)
-			}
-			break
-		case err := <-errs:
-			if e == nil || e.Error() != err.Error() {
-				t.Fatalf("expected error %v, got %v", e, err)
+	perpetualCheck:
```

Comment:
I'm not sure I understand the intent here - without the label and explicit `break`s wouldn't we have the same execution flows?

Reply:
I'm not sure, unless we replace `break`s with `return`s.
A "regular" `break` doesn't interrupt the outer loop:
```golang
for {
  select {
    case ...:
      // Stop `select`-ing only, skip to the next `for` iteration
      break
  }
}
```
therefore, with no labeled `break`, we have no way to exit cleanly from that function as it would:
- either time out and fail: `t.Fatalf("timed out waiting for value %d and error %d", expected, e)`.
- or fail in case of unexpected errors: `t.Fatalf("expected error %v, got %v", e, err)`

Reply:
You're in an anonymous function block, so prefer to use the `return` - labels and GOTO are not ideal.

Reply:
Also, if you `break` out of the `select`, the `for` has nothing else to do and you will go to another loop - so ... what is the difference there between breaking out of the `for` and re-running it (while only ever allowing it to have one iteration) and just running another iteration? That was my original comment.